src/ui/ActionFilters/McExceptionFilter.cs

```diff
@@ -3,41 +3,65 @@
 using Microsoft.ApplicationInsights;
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Microsoft.AspNetCore.Mvc.ViewFeatures;
+using Microsoft.Extensions.Logging;
 
 namespace GovUk.Education.ManageCourses.Ui.ActionFilters
 {
     public class McExceptionFilter : IExceptionFilter
     {
+        private readonly ILogger _logger;
+
+        public McExceptionFilter(ILogger<McExceptionFilter> logger)
+        {
+            _logger = logger;
+        }
+
         public void OnException(ExceptionContext context)
         {
             var swaggerException = context.Exception as SwaggerException;
-            
-            if (swaggerException?.StatusCode >= 400 && swaggerException?.StatusCode <= 404)
+
+            if (swaggerException?.StatusCode >= 400 && swaggerException.StatusCode <= 404)
             {
-                context.Result = new RedirectToActionResult("Index", "Error", new { statusCode = swaggerException?.StatusCode });
+                context.Result = StatusCodeViewResult(context, swaggerException.StatusCode);
```

Comment:
402 will become 500/server error?

Reply:
400?

Reply:
But we need to handle a 400

Reply:
@defong Yes, 400 & 402 from the api would result in server error page in the UI. But I don't think the api produces such status codes.
This is no change from existing behaviour.

The list of views returned per status code is in `Views/Error/Index.cshtml`, this can be extended should we want to handle them differently in the future.